yapf/__init__.py

```diff
@@ -121,6 +121,7 @@ def main(argv):
       parser.error('cannot use --in_place or --diff flags when reading '
                    'from stdin')
 
+    encoding, sys.stdin = py3compat.stdin()
```

Comment:
Personally I'm not really a fan of monkey patching sys.stdin outside of a contextmanager. I think this would be cleaner that way.